### ISSUE TITLE
attention_mask

### ISSUE THREAD

**Issue by LindgeW** [nguyenvulebinh/AV-HuBERT-S2S#1]
Hi, thanks for your nice work. I have a question about the attention_mask parameter in `model.generate`:

For the official api, https://huggingface.co/docs/transformers/model_doc/speech_to_text#transformers.Speech2TextModel.forward.attention_mask,
attention_mask (torch.Tensor of shape (batch_size, sequence_length), optional) — Mask to avoid performing attention on padding token indices. 
Mask values selected in [0, 1]:
  1 for tokens that are not masked,
  0 for tokens that are masked.

But in your implementation, attention_mask is an all-False matrix, which is confusing. How does your attention_mask work exactly?

Could you please explain this in detail?

**Comment by nguyenvulebinh** [comment]
I guess this is the part convert from bool to 1, 0 like you mentioned:

https://github.com/nguyenvulebinh/AV-HuBERT-S2S/blob/841efba840404582b5c8e0073b9599cc367e9c5b/src/model/avhubert.py#L542

**Comment by LindgeW** [comment]
Thanks for your reply. I tried to use the `AV-HuBERT-MuAViC-en` checkpoint to  predict LRS3 test data but the WER seems to be high (nearly 40%). Is this result normal?

**Comment by nguyenvulebinh** [comment]
I never test LRS3 before. Have you tried to use the original checkpoint from muavic repository? I only test with LRS2, the wer around 7.2.

**Comment by LindgeW** [comment]
Hi, `run_example.py` shows the single example inference. For efficiency, can we do inference with batch samples by padding (using `attention_mask`)?  I have tried it by padding (i.e., valid for False and padding for True in `attention_mask`), but obtained worse results on LRS3 (WER > 60%).  Strangely, when the `attention_mask`  is set to full-False, the results get better (WER = ~25%). 

for example, 4x5 `attention_mask`:
[False, False, True, True, True]
[False, False, False, False, False]
[False, False, False, True, True]
[False, False, False, False, True]
(True for padding)
